### PR TITLE
Added additional requester options for redirects

### PR DESCRIPTION
**Pull request by codenirvana**
Added:
- followOriginalHttpMethod
- maxRedirects
- removeRefererHeader

### CODE REVIEW

File: test/integration/requester-spec/redirect.test.js [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+            it('should preserve referer header set in the initial request', function () {
+                var response = testrun.response.getCall(0).args[2];
+
+                expect(response).to.have.property('code', 200);
+                expect(response).to.have.property('stream');
+                expect(response.stream.toString()).to.equal('okay');
+
+                expect(hits).to.have.lengthOf(2);
+                expect(hits[1]).to.have.property('url', '/');
```
Comment (kunagpal) [inline test/integration/requester-spec/redirect.test.js]: See comment above.

File: test/integration/requester-spec/redirect.test.js [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+        });
+
+        it('should follow maximum 10 redirects by default and throw error after that', function () {
+            var response = testrun.response.getCall(0);
+
+            expect(response.args[0]).to.have.property('message',
+                `Exceeded maxRedirects. Probably stuck in a redirect loop ${HOST}/1`);
+
+            expect(hits).to.have.lengthOf(11);
+            expect(hits[10]).to.have.property('url', '/1');
```
Comment (kunagpal) [inline test/integration/requester-spec/redirect.test.js]: See comment above.

File: test/integration/requester-spec/redirect.test.js [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+        it('should follow all the redirects with maxRedirects set', function () {
+            var response = testrun.response.getCall(0).args[2];
+
+            expect(response).to.have.property('code', 200);
+            expect(response).to.have.property('stream');
+            expect(response.stream.toString()).to.equal('okay');
+
+            expect(hits).to.have.lengthOf(12);
+            expect(hits[11]).to.have.property('url', '/');
```
Comment (kunagpal) [inline test/integration/requester-spec/redirect.test.js]: See comment above.

File: test/integration/requester-spec/redirect.test.js [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+        hits = [],
+        PORT = 5050,
+        HOST = 'http://localhost:' + PORT;
+
+    before(function (done) {
+        server = http.createServer(function (req, res) {
+            var hops;
+
+            // keep track of all the requests made during redirects.
+            hits.push({
```
[the remark this answers is not in the document]
Comment (codenirvana) [reply test/integration/requester-spec/redirect.test.js]: Only extracting properties needed for assertions later.

**Pull request merged**